src/Libraries/RevitNodesUI/GenericClasses.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Autodesk.DesignScript.Geometry;
+using Autodesk.DesignScript.Runtime;
+using DSRevitNodesUI;
+using RVT = Autodesk.Revit.DB;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+
+using Dynamo.Utilities;
+using Dynamo.Models;
+using ProtoCore.AST.AssociativeAST;
+
+namespace DSRevitNodesUI
+{
+
+    /// <summary>
+    /// Generic UI Dropdown node baseclass for Revit Elements.
+    /// This class populates a dropdown with all Revit elements of the specified type.
+    /// It uses a filtered element collector filtering by class.
+    /// </summary>
+    public abstract class CustomRevitElementDropDown : RevitDropDownBase
+    {
+        /// <summary>
+        /// Generic Revit Element Class Dropdown Node
+        /// </summary>
+        /// <param name="name">Name of the Node</param>
+        /// <param name="elementType">Type of Revit Element to display</param>
+        public CustomRevitElementDropDown(string name, Type elementType) : base(name) { this.ElementType = elementType; PopulateItems(); }
+
+        /// <summary>
+        /// Constant message for missing Types
+        /// </summary>
+        private const string noTypes = "No Types available.";
+
+        /// <summary>
+        /// Type of Element
+        /// </summary>
+        public Type ElementType;
+
+        protected override CoreNodeModels.DSDropDownBase.SelectionState PopulateItemsCore(string currentSelection)
+        {
+            PopulateItems();
+            return SelectionState.Done;
+        }
+        /// <summary>
+        /// Populate the Dropdown menu
+        /// </summary>
+        public void PopulateItems()
+        {
+            if (this.ElementType != null)
+            {
+                // Clear the Items
+                Items.Clear();
+
+                // Set up a new element collector using the Type field
+                var fec = new RVT.FilteredElementCollector(DocumentManager.Instance.CurrentDBDocument).OfClass(ElementType);
+
+                // If there is nothing in the collector add the missing Type message to the Dropdown menu.
+                if (fec.ToElements().Count == 0)
+                {
+                    Items.Add(new CoreNodeModels.DynamoDropDownItem(noTypes, null));
+                    SelectedIndex = 0;
+                    return;
+                }
+
+                // if the elementtype is a RebarHookType add an initial "None" value which does not come from the collector
+                if (this.ElementType.FullName == "Autodesk.Revit.DB.Structure.RebarHookType") Items.Add(new CoreNodeModels.DynamoDropDownItem("None", null));
```

Comment:
@mjkkirschner , the Rebar constructor accepts null as a hook type value and translates this into a rebar hook type called "none". But querying the existing hook types won't give you "none" as an existing hook type since its just null. I don't know if there are other constructors accepting null as well.
I could drop it entirely or always add a "none" value returning null. What would make more sense?